protocol-units/bridge/chains/movement/src/utils.rs

```diff
@@ -0,0 +1,170 @@
+use crate::MovementClient;
+use anyhow::{Context, Result};
+use aptos_sdk::{
+	crypto::ed25519::Ed25519Signature,
+	move_types::language_storage::TypeTag,
+	move_types::{ident_str, language_storage::ModuleId},
+	rest_client::aptos_api_types::{
+		EntryFunctionId, MoveType, Transaction as AptosTransaction, TransactionInfo, ViewRequest,
+	},
+	rest_client::Client as RestClient,
+	transaction_builder::TransactionFactory,
+	types::{
+		account_address::AccountAddress,
+		chain_id::ChainId,
+		transaction::{EntryFunction, SignedTransaction, TransactionPayload},
+		LocalAccount,
+	},
+};
+use derive_new::new;
+use keccak_hash::H256;
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+use thiserror::Error;
+
+#[derive(Debug, Error, Clone, PartialEq, Eq)]
+pub enum MovementAddressError {
+	#[error("Invalid hex string")]
+	InvalidHexString,
+	#[error("Invalid byte length for AccountAddress")]
+	InvalidByteLength,
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
+pub struct MovementAddress(pub AccountAddress);
+
+impl From<Vec<u8>> for MovementAddress {
+	fn from(vec: Vec<u8>) -> Self {
+		// Ensure the vector has the correct length
+		assert_eq!(vec.len(), AccountAddress::LENGTH);
+
+		let account_address =
+			AccountAddress::from_bytes(vec).expect("Invalid byte length for AccountAddress");
+		MovementAddress(account_address)
+	}
+}
+
+impl From<&str> for MovementAddress {
+	fn from(s: &str) -> Self {
+		let s = s.trim_start_matches("0x");
+		let bytes = hex::decode(s).expect("Invalid hex string");
+		bytes.into()
+	}
+}
+
+/// limit of gas unit
+const GAS_UNIT_LIMIT: u64 = 100000;
+/// minimum price of gas unit of aptos chains
+pub const GAS_UNIT_PRICE: u64 = 100;
+
+/// Wrapper struct that adds indexing information to a type
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, new)]
+pub struct Indexed<T> {
+	inner: T,
+	#[new(default)]
+	/// Optional sequence data that is useful during indexing
+	pub sequence: Option<u32>,
+}
+
+/// Send Aptos Transaction
+pub async fn send_aptos_transaction(
+	rest_client: &RestClient,
+	signer: &mut LocalAccount,
+	payload: TransactionPayload,
+) -> Result<AptosTransaction> {
```

Comment:
Avoid using anyhow::Error in libs

Reply:
Yes again, mostly just placeholder code to be handled properly in this issue https://github.com/movementlabsxyz/movement/issues/224#issue-2413712253

I think its ok to merge some of this placeholder code into the feature branch (but not into main!)